### PR TITLE
Changes for running the docker tutorials successfully:

### PR DESCRIPTION
**Pull request by NicoletaOita**
- updated docker compose file for activemq to create the link between the gateway and the broker
- enabled some needed features at gateway startup (redis, http.proxy)

### CODE REVIEW

File: JMS/activemq/docker-compose.yml [hunk 1]
```diff
@@ -5,5 +5,9 @@ services:
     ports:
       - "8000:8000"
     hostname: "kaazing.example.com"
+    links: 
+      - activemq:activemq_activemq
   activemq:
     build: broker
+    hostname: "kaazing.example.com"
```
Comment (AdrianCozma) [inline JMS/activemq/docker-compose.yml]: why is this needed?

File: enterprise-shield/AMQP/rabbitmq/docker-compose.yml [hunk 2]
```diff
@@ -5,7 +5,11 @@ services:
     ports:
       - "8000:8000"
     hostname: "kaazing.example.com"
+    environment:
+      GATEWAY_OPTS: -Dfeature.http.proxy=true -Djava.naming.provider.url=FILE://kaazing-gateway/bin"
   internal.example.com: 
     build: internal
+    environment:
+      GATEWAY_OPTS: -Dfeature.http.proxy=true -Djava.naming.provider.url=FILE://kaazing-gateway/bin"
```
Comment (dpwspoon) [inline enterprise-shield/AMQP/rabbitmq/docker-compose.yml]: Why is the java.naming.provdier.url?  I wouldn't expect the .bindings file to be in bin.  And I don't think this would be needed for amqp service

Comment (NicoletaOita) [inline enterprise-shield/AMQP/rabbitmq/docker-compose.yml]: It wasn't needed, I forgot to remove it.

File: JMS/activemq/docker-compose.yml (hunk 1 above)
Comment (NicoletaOita) [inline JMS/activemq/docker-compose.yml]: I added it for the links, but removed it now, together with the links.

**Comment by NicoletaOita** [comment]
@dpwspoon I've removed the links and the java naming provider url. I've also changed the gateway versions, added the necessary jars and commented the git clone part that returns an error for the javascript tutorials, such that you can see the exact changes that I've referred to in my [previous comment] (https://github.com/kaazing/QA-Roadmap/issues/35#issuecomment-238810771). Please let me know what you think.

**Comment by dpwspoon** [comment]
This looks good to me.  I haven't / won't get a chance to run them.  @AdrianCozma please feel free to merge

**Comment by dpwspoon** [comment]
Thanks @NicoletaOita

**Comment by NicoletaOita** [comment]
@AdrianCozma I've replaced the old demos that used git with the already build ones from github, as @dpwspoon suggested [here] (https://github.com/kaazing/QA-Roadmap/issues/35#issuecomment-239034084). Please review.

File: AMQP/rabbitmq/gateway/Dockerfile [hunk 3]
```diff
@@ -23,15 +22,10 @@ RUN set -ex \
...
-RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" \
-  && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
-  && gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc \
-  && grep " node-v$NODE_VERSION-linux-x64.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
-  && tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 \
-  && rm "node-v$NODE_VERSION-linux-x64.tar.xz" SHASUMS256.txt.asc SHASUMS256.txt
-
-RUN git clone https://github.com/kaazing/javascript.client.tutorials ./web/javascript.client.tutorials
-RUN npm --prefix ./web/javascript.client.tutorials/amqp install ./web/javascript.client.tutorials/amqp 
+RUN curl -LOk  https://github.com/kaazing/javascript.client.tutorials/files/324464/websocket-jms-tutorial.zip
```
Comment (dpwspoon) [inline AMQP/rabbitmq/gateway/Dockerfile]: Nit: Best practices in docker is to combine this all of these into one "RUN" command to minimize layers, see how it is done on the left via &&.  Also ideally we would check the hash one it to be super secure

**Comment by dpwspoon** [comment]
Looks like progress to me :-)

File: AMQP/rabbitmq/gateway/Dockerfile (hunk 3 above)
Comment (NicoletaOita) [inline AMQP/rabbitmq/gateway/Dockerfile]: @dpwspoon I've changed the Dockerfiles such that there is a single RUN command for downloading and unzipping the tutorials to the right locations.
Regarding the check to be secure, I assume you referred to the -k flag that I added, which doesn't check for the certificate, but if I don't use it, the tutorial is not downloaded.

**Comment by AdrianCozma** [comment]
looks OK to me

**Comment by dpwspoon** [comment]
Typically we build or create sha1 or .asc at build/release time.  I don’t think we generate them for the client tutorials, thus it isn’t needed for this PR.   Ideally we would already be generating them and you could use them.  There is a small possibility that the github API provides a way to get them.  Note: I suggested it as it is listed as best practice in security here: https://github.com/docker-library/official-images

**Comment by NicoletaOita** [comment]
I've removed the sha1 check, please review.

File: JMS/activemq/docker-compose.yml [hunk 4]
```diff
@@ -6,4 +6,4 @@ services:
       - "8000:8000"
     hostname: "kaazing.example.com"
   activemq:
-    build: broker
+    image: webcenter/activemq:latest
```
Comment (AdrianCozma) [inline JMS/activemq/docker-compose.yml]: I think we should use 5.10.0 instead of latest here

Comment (NicoletaOita) [inline JMS/activemq/docker-compose.yml]: @AdrianCozma If I replace the latest version with 5.10.0, I get the following error at gateway start:
```
gateway_1   | INFO  Started server successfully in 1.516 secs at 2016-08-18 11:37:02
gateway_1   | Exception in thread "ActiveMQ Connection Executor: tcp://activemq/172.18.0.2:61616@43328" java.lang.NullPointerException
gateway_1   |   at org.apache.activemq.ActiveMQConnection$5.run(ActiveMQConnection.java:2005)
gateway_1   |   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
gateway_1   |   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
gateway_1   |   at java.lang.Thread.run(Thread.java:745)
```

Comment (NicoletaOita) [inline JMS/activemq/docker-compose.yml]: I've filed a new [issue] (https://github.com/kaazing/tickets/issues/702) for the above mentioned behavior.
